diff --git a/CMakeLists.txt b/CMakeLists.txt
index c27140d..87bd66e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -51,64 +51,64 @@ target_link_libraries(rkai_VLM_Demo
 install(TARGETS rkai_LLM_Demo rkai_VLM_Vision_Demo rkai_VLM_Demo DESTINATION bin)
 install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/app_conf DESTINATION share)
 
-# 在下载模型文件之前添加目录创建
-file(MAKE_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/app_conf/models)
-
-if(${ROCKCHIP_AI_CHIP} STREQUAL "rv1126b")
-    set(MODEL_FILE  "${CMAKE_CURRENT_SOURCE_DIR}/app_conf/models/Qwen3-0.6B-w4a16.rv1126b.rkllm")
-    set(MODEL_URL "https://www.modelscope.cn/models/eliasning/Qwen3-0.6B-w4a16.rv1126b.rkllm/resolve/master/Qwen3-0.6B-w4a16.rv1126b.rkllm")
-
-    # 仅当文件不存在才下载
-    if(NOT EXISTS ${MODEL_FILE})
-        find_program(WGET wget)          # 先找 wget
-        if(NOT WGET)
-            message(FATAL_ERROR "wget not found, please install wget or enable CMake SSL")
-        endif()
-        execute_process(
-            COMMAND ${WGET} -O ${MODEL_FILE} ${MODEL_URL}
-            RESULT_VARIABLE RET
-            TIMEOUT 300)
-        if(NOT RET EQUAL 0)
-            file(REMOVE ${MODEL_FILE})   # 下载失败时清掉残片
-            message(FATAL_ERROR "wget download failed: ${RET}")
-        endif()
-    endif()
-
-    set(MODEL_FILE  "${CMAKE_CURRENT_SOURCE_DIR}/app_conf/models/fastvlm-0.5b-fp16.rv1126b.rknn")
-    set(MODEL_URL "https://www.modelscope.cn/models/eliasning/Qwen3-0.6B-w4a16.rv1126b.rkllm/resolve/master/fastvlm-0.5b-fp16.rv1126b.rknn")
-
-    # 仅当文件不存在才下载
-    if(NOT EXISTS ${MODEL_FILE})
-        find_program(WGET wget)          # 先找 wget
-        if(NOT WGET)
-            message(FATAL_ERROR "wget not found, please install wget or enable CMake SSL")
-        endif()
-        execute_process(
-            COMMAND ${WGET} -O ${MODEL_FILE} ${MODEL_URL}
-            RESULT_VARIABLE RET
-            TIMEOUT 300)
-        if(NOT RET EQUAL 0)
-            file(REMOVE ${MODEL_FILE})   # 下载失败时清掉残片
-            message(FATAL_ERROR "wget download failed: ${RET}")
-        endif()
-    endif()
-
-    set(MODEL_FILE  "${CMAKE_CURRENT_SOURCE_DIR}/app_conf/models/fastvlm-0.5B-w8a8_level0.rv1126b.rkllm")
-    set(MODEL_URL "https://www.modelscope.cn/models/eliasning/Qwen3-0.6B-w4a16.rv1126b.rkllm/resolve/master/fastvlm-0.5B-w8a8_level0.rv1126b.rkllm")
-
-    # 仅当文件不存在才下载
-    if(NOT EXISTS ${MODEL_FILE})
-        find_program(WGET wget)          # 先找 wget
-        if(NOT WGET)
-            message(FATAL_ERROR "wget not found, please install wget or enable CMake SSL")
-        endif()
-        execute_process(
-            COMMAND ${WGET} -O ${MODEL_FILE} ${MODEL_URL}
-            RESULT_VARIABLE RET
-            TIMEOUT 300)
-        if(NOT RET EQUAL 0)
-            file(REMOVE ${MODEL_FILE})   # 下载失败时清掉残片
-            message(FATAL_ERROR "wget download failed: ${RET}")
-        endif()
-    endif()
-endif()
+# # 在下载模型文件之前添加目录创建
+# file(MAKE_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/app_conf/models)
+
+# if(${ROCKCHIP_AI_CHIP} STREQUAL "rv1126b")
+#     set(MODEL_FILE  "${CMAKE_CURRENT_SOURCE_DIR}/app_conf/models/Qwen3-0.6B-w4a16.rv1126b.rkllm")
+#     set(MODEL_URL "https://www.modelscope.cn/models/eliasning/Qwen3-0.6B-w4a16.rv1126b.rkllm/resolve/master/Qwen3-0.6B-w4a16.rv1126b.rkllm")
+
+#     # 仅当文件不存在才下载
+#     if(NOT EXISTS ${MODEL_FILE})
+#         find_program(WGET wget)          # 先找 wget
+#         if(NOT WGET)
+#             message(FATAL_ERROR "wget not found, please install wget or enable CMake SSL")
+#         endif()
+#         execute_process(
+#             COMMAND ${WGET} -O ${MODEL_FILE} ${MODEL_URL}
+#             RESULT_VARIABLE RET
+#             TIMEOUT 300)
+#         if(NOT RET EQUAL 0)
+#             file(REMOVE ${MODEL_FILE})   # 下载失败时清掉残片
+#             message(FATAL_ERROR "wget download failed: ${RET}")
+#         endif()
+#     endif()
+
+#     set(MODEL_FILE  "${CMAKE_CURRENT_SOURCE_DIR}/app_conf/models/fastvlm-0.5b-fp16.rv1126b.rknn")
+#     set(MODEL_URL "https://www.modelscope.cn/models/eliasning/Qwen3-0.6B-w4a16.rv1126b.rkllm/resolve/master/fastvlm-0.5b-fp16.rv1126b.rknn")
+
+#     # 仅当文件不存在才下载
+#     if(NOT EXISTS ${MODEL_FILE})
+#         find_program(WGET wget)          # 先找 wget
+#         if(NOT WGET)
+#             message(FATAL_ERROR "wget not found, please install wget or enable CMake SSL")
+#         endif()
+#         execute_process(
+#             COMMAND ${WGET} -O ${MODEL_FILE} ${MODEL_URL}
+#             RESULT_VARIABLE RET
+#             TIMEOUT 300)
+#         if(NOT RET EQUAL 0)
+#             file(REMOVE ${MODEL_FILE})   # 下载失败时清掉残片
+#             message(FATAL_ERROR "wget download failed: ${RET}")
+#         endif()
+#     endif()
+
+#     set(MODEL_FILE  "${CMAKE_CURRENT_SOURCE_DIR}/app_conf/models/fastvlm-0.5B-w8a8_level0.rv1126b.rkllm")
+#     set(MODEL_URL "https://www.modelscope.cn/models/eliasning/Qwen3-0.6B-w4a16.rv1126b.rkllm/resolve/master/fastvlm-0.5B-w8a8_level0.rv1126b.rkllm")
+
+#     # 仅当文件不存在才下载
+#     if(NOT EXISTS ${MODEL_FILE})
+#         find_program(WGET wget)          # 先找 wget
+#         if(NOT WGET)
+#             message(FATAL_ERROR "wget not found, please install wget or enable CMake SSL")
+#         endif()
+#         execute_process(
+#             COMMAND ${WGET} -O ${MODEL_FILE} ${MODEL_URL}
+#             RESULT_VARIABLE RET
+#             TIMEOUT 300)
+#         if(NOT RET EQUAL 0)
+#             file(REMOVE ${MODEL_FILE})   # 下载失败时清掉残片
+#             message(FATAL_ERROR "wget download failed: ${RET}")
+#         endif()
+#     endif()
+# endif()
diff --git a/rkai_VLM/src/rkai_vlm_vision.cpp b/rkai_VLM/src/rkai_vlm_vision.cpp
index 3009d6c..d6c6765 100644
--- a/rkai_VLM/src/rkai_vlm_vision.cpp
+++ b/rkai_VLM/src/rkai_vlm_vision.cpp
@@ -163,7 +163,7 @@ int rkai_vlm_vision_init(const char *model_path, rkaiCtxVision *rkai_ctx) {
 #endif
 
     ret = rknn_init(&ctx, (void *)model_path, 0, 0, NULL);
-    free(model);
+    // free(model);
     if (ret < 0) {
         printf("rknn_init fail! ret=%d\n", ret);
         return -1;
diff --git a/samples/rkai_samples_llm.cpp b/samples/rkai_samples_llm.cpp
index 32f8ef5..c6166a8 100644
--- a/samples/rkai_samples_llm.cpp
+++ b/samples/rkai_samples_llm.cpp
@@ -154,7 +154,6 @@ int main(int argc, char **argv) {
             strcpy(input_buf, pre_input[idx]);
             printf("%s\n", input_buf);
         }
-
         rkai_llm_run(ctx, input_buf);
     }
 
diff --git a/samples/rkai_samples_vlm.cpp b/samples/rkai_samples_vlm.cpp
index 5546074..e972bd3 100644
--- a/samples/rkai_samples_vlm.cpp
+++ b/samples/rkai_samples_vlm.cpp
@@ -106,19 +106,44 @@ static int load_config(const char *filename, rkaiVLMCfg *cfg) {
     return 0;
 }
 
-
+static char *join_path(const char *dir, const char *file)
+{
+    if (!dir || !file) {
+        return NULL;
+    } 
+
+    int need_sep = (dir[strlen(dir) - 1] != '/');
+    size_t len   = strlen(dir) + strlen(file) + need_sep + 1; /* +1 for '\0' */
+    char  *full  = (char *)malloc(len*sizeof(char));
+    if (full) {
+        snprintf(full, len, "%s%s%s", dir, need_sep ? "/" : "", file);
+    }
+    return full;
+}
 
 int main(int argc, char **argv) {
-    if ((argc != 2) || (strcmp(argv[1], "--help") == 0) || (strcmp(argv[1], "-h") == 0)) { 
-        printf("Usage: %s <config_json>\n", argv[0]);
-        printf("Such as: rkai_VLM_Demo /usr/share/app_conf/config/rkai_vlm_fastvlm_rv1126b.json\n");
+    if ((argc != 3) || (strcmp(argv[1], "--help") == 0) || (strcmp(argv[1], "-h") == 0)) { 
+        printf("Usage: %s <config_path> <json_file>\n", argv[0]);
+        printf("Such as: rkai_VLM_Demo /usr/share/app_conf/ rkai_vlm_fastvlm_rv1126b.json\n");
         return -1;
     }
 
+    printf("app_conf path:%s,conf_file:%s\n",argv[1],argv[2]);
     rkai_vlm_get_version();
-    const char *cfg_file = argv[1];
+    const char *app_conf_path = argv[1];
+    const char *cfg_file = argv[2];
+    
+    const char *config_path = argv[1];
+    const char *json_file   = argv[2];
+
+    char *cfg_full_file  = join_path(app_conf_path,json_file);
+    if(cfg_full_file == NULL){
+        printf("Cannot get full config json file,Please check.\n");
+        return -1;
+    }
+
     rkaiVLMCfg config;
-    if (load_config(cfg_file, &config) == 0) {
+    if (load_config(cfg_full_file, &config) == 0) {
         printf("==================Config Loaded:=================\n");
         printf(" vision_model_path: %s\n", config.visionModelPath);
         printf(" llm_model_path: %s\n", config.llmModelPath);
@@ -136,7 +161,9 @@ int main(int argc, char **argv) {
         printf("Failed to load config.\n");
         return -1;
     }
+    
 
+    #if 0
     char input_path[128];
     char vision_model_path[128];
     char llm_model_path[128];
@@ -149,6 +176,20 @@ int main(int argc, char **argv) {
 
     strncpy(llm_model_path, config.llmModelPath, sizeof(llm_model_path) - 1);
     llm_model_path[sizeof(llm_model_path) - 1] = '\0';
+    #else
+    char input_path[128];
+    char vision_model_path[128];
+    char llm_model_path[128];
+
+    strncpy(input_path, config.inputPath, sizeof(input_path) - 1);
+    input_path[sizeof(input_path) - 1] = '\0'; 
+
+    strncpy(vision_model_path, config.visionModelPath, sizeof(vision_model_path) - 1);
+    vision_model_path[sizeof(vision_model_path) - 1] = '\0';
+
+    strncpy(llm_model_path, config.llmModelPath, sizeof(llm_model_path) - 1);
+    llm_model_path[sizeof(llm_model_path) - 1] = '\0';
+    #endif
 
     int ret;
     rkaiCtxVision *rkai_ctx = (rkaiCtxVision *)malloc(sizeof(rkaiCtxVision));
@@ -157,7 +198,6 @@ int main(int argc, char **argv) {
     rkaiCtxVLM *llm_ctx = (rkaiCtxVLM *)malloc(sizeof(rkaiCtxVLM));
     memset(llm_ctx,0,sizeof(rkaiCtxVLM));
     
-
     rkaiTimer t_ms;
     tik(&t_ms);
     ret = rkai_vlm_vision_init(vision_model_path, rkai_ctx);
@@ -281,5 +321,11 @@ int main(int argc, char **argv) {
         vtol = NULL;
     }
 
+    if(cfg_full_file) {
+        free(cfg_full_file);
+        cfg_full_file = NULL;
+    }
+
+
     return 0;
 }
\ No newline at end of file
